pandas/core/generic.py

```diff
@@ -7011,7 +7023,8 @@ def asof(self, where, subset=None):
 
         # mask the missing
         missing = locs == -1
-        data = self.take(locs, is_copy=False)
+        d = self.take(locs)
+        data = d.copy()
```

Comment:
Were there test failures if you didn't do this `d.copy()` step?

Reply:
Yes, it was breaking builds with SettingwithCopyError https://github.com/pandas-dev/pandas/pull/30615#discussion_r363065280